core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -24,4 +26,31 @@
  */
 public abstract class FileInStream extends InputStream implements BoundedStream, PositionedReadable,
     Seekable {
+  /**
+   * Reads up to len bytes of data from the input stream into the byte buffer.
+   *
+   * @param byteBuffer the buffer into which the data is read
+   * @param off the start offset in the buffer at which the data is written
+   * @param len the maximum number of bytes to read
+   * @return the total number of bytes read into the buffer, or -1 if there is no more
+   *         data because the end of the stream has been reached
+   */
+  public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
+    int nread = 0;
+    int rd = 0;
+    final int sz = len;
+    final byte[] dest = new byte[sz];
+    while (rd >= 0 && nread < sz) {
+      rd = read(dest, nread, sz - nread);
+      if (rd >= 0) {
+        nread += rd;
+      }
+    }
+    if (nread == -1) { // EOF
+      nread = 0;
```

Comment:
These lines don't look right: `nread` is only updated by adding `rd`, where `rd` is non-negative. `nread` is 0 at the beginning. Therefore `nread` is non-negative. Why do we check if it is -1?

Reply:
@dbw9580 Good catch! This code is obtained from AlluxioFuseFileSystem.java->readInternal, so it seems both of them have similar problems, can you fix this?

Reply:
This line was added in this commit https://github.com/Alluxio/alluxio/commit/0d30732ba71674357d9d30c3c34fa701efaacfda so it's almost been here since like genesis. 🤦

Reply:
And this is indeed an obvious mistake!